compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java

```diff
@@ -398,6 +405,30 @@ private void findTypeParamInTupleForArray(DiagnosticPos pos, BArrayType expType,
         findTypeParam(pos, expType.eType, tupleElementType, env, resolvedTypes, result);
     }
 
+    private void findTypeParamInUnionForArray(DiagnosticPos pos, BArrayType expType, BUnionType actualType,
+                                              SymbolEnv env, HashSet<BType> resolvedTypes, FindTypeParamResult result) {
+        LinkedHashSet<BType> tupleTypes = new LinkedHashSet<>();
+        for (BType type : actualType.getMemberTypes()) {
+            if (type.tag == TypeTags.ARRAY) {
```

Comment:
What about tuples, and similarly records for maps?

Reply:
I have fixed this please help review.
Records: https://github.com/ballerina-platform/ballerina-lang/pull/22346/commits/d617fd7cae953123c47176d6b7bbb38f2d3fecae
Tuples: https://github.com/ballerina-platform/ballerina-lang/pull/22346/commits/12fc9860180fb75f5c7ff449ac5779cd1a6b1a27